src/kfmon.c

```diff
@@ -382,49 +334,35 @@ int nm_kfmon_list_request(const char *restrict foo __attribute__((unused))) {
     }
 
     // Attempt to send the specified command in full over the wire
-    status = send_ipc_command(data_fd, "list", NULL);   // FIXME: switch to gui-list
+    status = send_ipc_command(data_fd, ipc_cmd, NULL);
     // If it failed, return early, after closing the socket
     if (status != EXIT_SUCCESS) {
         close(data_fd);
         return status;
     }
 
-    // We'll want to retrieve our watch list in there.
-    kfmon_watch_list_t list = { 0 };
-
     // We'll be polling the socket for a reply, this'll make things neater, and allows us to abort on timeout,
     // in the unlikely event there's already an IPC session being handled by KFMon,
     // in which case the reply would be delayed by an undeterminate amount of time (i.e., until KFMon gets to it).
     // Here, we'll want to timeout after 2s
     ipc_handler_t handler = &handle_list_reply;
-    status = wait_for_replies(data_fd, 500, 4, handler, (void *) &list);
+    status = wait_for_replies(data_fd, 500, 4, handler, (void *) list);
     // NOTE: We happen to be done with the connection right now.
     //       But if we still needed it, KFMON_IPC_POLL_FAILURE would warrant an early abort w/ a forced close().
 
-    // Walk the list
-    NM_LOG("List has %zu nodes", list.count);
-    NM_LOG("Head is at %p", list.head);
-    NM_LOG("Tail is at %p", list.tail);
-    for (kfmon_watch_node_t* node = list.head; node != NULL; node = node->next) {
-        NM_LOG("Dumping node at %p", node);
-        NM_LOG("idx: %hhu // filename: %s // label: %s", node->watch.idx, node->watch.filename, node->watch.label);
-    }
-
-    // Destroy it
-    teardown_list(&list);
-
     // Bye now!
     close(data_fd);
     return status;
 }
 
 // Giant ladder of fail
-nm_action_result_t* nm_kfmon_return_handler(kfmon_ipc_errno_e status, char **err_out) {
+void* nm_kfmon_error_handler(kfmon_ipc_errno_e status, char **err_out) {
     #define NM_ERR_RET NULL
 
     switch (status) {
+        // NOTE: Should never be passed a succes status code!
         case KFMON_IPC_OK:
-            NM_RETURN_OK(nm_action_result_silent());
+            return NULL;
```

Comment:
I know this case will almost never be reached, but it would be nice if you used `NM_RETURN_OK` with NULL to set err to NULL (most callers don't initialize it explicitly) on success to prevent segfaulting if it is use.